test/Streamly/Test/Prelude/Concurrent.hs

```diff
@@ -60,6 +65,23 @@ mvarSequenceOp mv n x = do
     else dbgMVar ("put mvarSequenceOp" <> msg)
             (replicateM_ (fromIntegral n) (putMVar mv ())) >> return x
 
+checkCleanup :: IsStream t
+    => Int
+    -> (t IO Int -> SerialT IO Int)
+    -> (t IO Int -> t IO Int)
+    -> IO ()
+checkCleanup d t op = do
```

Comment:
This is duplicated. It exists in `Fold` as well.

Reply:
and `Parallal`

Reply:
Should be refactored into the Prelude module